src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -361,6 +363,14 @@ public void setFareService(FareService fareService) {
     this.fareService = fareService;
   }
 
+  public EmissionsService getEmissionsService() {
+    return emissionsService;
+  }
+
+  public void setEmissionsService(EmissionsService emissionsService) {
+    this.emissionsService = emissionsService;
+  }
+
```

Comment:
Sorry, we do not pass "stuff" around in the graph any more, you need to inject it using Dagger.

Reply:
Thanks for putting time into reviewing this draft! There are alot of changes coming soon, but you have pointed out some things I hadn't considered. I shall fix these as well before updating the pr.

Reply:
Unfortunately, I'm having slight issues with the injection, as I am not sure where exactly it should be done. As far as I understand, we need to add the service/filter to the builder in RouteRequestToFilterChainMapper or ItineraryListChainBuilder, but we cannot inject anything in those. Could you maybe clarify how exactly a filter should be added without the graph? Thanks!

Reply:
I'm afraid you might have to spell this out for me, as I'm failing to see how to inject the service into the SerializedGraphObject without doing something strange. The SerializedGraphObject is initialised in the OTPMain only and I doubt you meant the emissionsservice is supposed to added to the SerializedGraphObject in the OPTMain. Also, I managed to inject and get the service from the OtpServerRequestContext, but of course it is a new instance and not the one handled during the building phase that contains the emissions data. So, I'm also failing to see how the service instance (that should be) injected into the SerializedGraphObject, will end up being the same instance that is gained from the context in the RouteRequestFilterChainMapper (as opposed to the brand new singleton without emissions data that I get now). Does there happen to be any existing service that has been implemented in the manner you are suggesting that I could maybe look at? I was only able to find the ones made in the old way and one that doesn't require the build phase at all (rideHailingService). Thanks!

Reply:
I don't think there is a full example for a service, so you're breaking new ground, but the worldEnveloperRepository and dataImportIssueSummary to combine data from various graph building phases and put that into the `SerializedGraphObject`.

https://github.com/opentripplanner/OpenTripPlanner/blob/bf6e4beb4e15914bc047d4428fe0d54606ca0337/src/main/java/org/opentripplanner/standalone/OTPMain.java#L148-L155